SPEECH.md

```diff
@@ -0,0 +1,369 @@
+
+After your Custom Speech project is set up and a model is published to a deployment endpoint, in the "Deployment" tab, save the "Endpoint ID".
+
+You will then need to modify your integration code as below.
+
+```diff
+  renderWebChat({
+    directLine: createDirectLine({
+      secret: 'YOUR_DIRECT_LINE_SECRET'
+    }),
+    language: 'en-US',
+    webSpeechPonyfillFactory: await createCognitiveServicesSpeechServicesPonyfillFactory({
+      region: 'YOUR_REGION',
++     speechRecognitionEndpointId: 'YOUR_ENDPOINT_ID',
+      subscriptionKey: 'YOUR_SUBSCRIPTION_KEY'
+    })
+  }, document.getElementById('webchat'));
+```
+
+### Custom Voice
+
+Custom Voice is a trained synthesis model for providing your user with an unique synthesized voice when performing text-to-speech.
+
+First, you need to set up a Custom Voice project. Please follow [this article to create a new Custom Voice project][Get started with Custom Voice].
+
+After your Custom Voice project is set up and a model is published to a deployment endpoint, in the "Deployment" tab, save the "Model / Voice name" and "Endpoint URL".
+
+You will then need to modify your integration code as below. The `selectVoice` function will be used to choose which trained synthesis model to use.
+
+```diff
+  renderWebChat({
+    directLine: createDirectLine({
+      secret: 'YOUR_DIRECT_LINE_SECRET'
+    }),
+    language: 'en-US',
++   selectVoice: () => ({ voiceURI: 'YOUR_VOICE_MODEL_NAME' }),
+    webSpeechPonyfillFactory: await createCognitiveServicesSpeechServicesPonyfillFactory({
+      region: 'YOUR_REGION',
++     speechSynthesisDeploymentId: 'YOUR_DEPLOYMENT_ID',
+      subscriptionKey: 'YOUR_SUBSCRIPTION_KEY'
+    })
+  }, document.getElementById('webchat'));
+```
+
+> Note: if you send SSML instead of plain text, make sure the voice model is correctly selected in your SSML.
+
+### Text-to-speech audio format
+
+To conserve bandwidth, you can set the text-to-speech audio format to one that consume less bandwidth by modifying your integration code as below.
+
+```diff
+  renderWebChat({
+    directLine: createDirectLine({
+      secret: 'YOUR_DIRECT_LINE_SECRET'
+    }),
+    language: 'en-US',
+    webSpeechPonyfillFactory: await createCognitiveServicesSpeechServicesPonyfillFactory({
+      region: 'YOUR_REGION',
++     speechSynthesisOutputFormat: 'audio-24khz-48kbitrate-mono-mp3',
+      subscriptionKey: 'YOUR_SUBSCRIPTION_KEY'
+    })
+  }, document.getElementById('webchat'));
+```
+
+Please refer to [this article for list of supported audio formats](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/rest-text-to-speech#audio-outputs).
+
+### Text normalization options
+
+Text normalization is an option to modify how the speech engine normalize texts. For example, when the user say, "I would like to order 2 4-piece of chicken nuggets." It could be recognized as "two four piece" (default) or "2 four piece" (inverse text normalization, or ITN).
```

Comment:
```suggestion
Text normalization is an ability to modify how the speech engine normalizes text. For example, when the user says, "I would like to order 2 4-piece chicken nuggets." It could be recognized as "two four piece" (default) or "2 four piece" (inverse text normalization, or ITN).
```